### ISSUE TITLE
Generated wrong post link when website is in a subdirectory

### ISSUE THREAD

**Issue by Bitrhythm** [theme-next/hexo-theme-next#473]
### I agree and want to create new issue 

- [X] Yes, I was on [Hexo Docs page](https://hexo.io/docs/), especially on [Templates](https://hexo.io/docs/templates.html), [Variables](https://hexo.io/docs/variables.html), [Helpers](https://hexo.io/docs/helpers.html) and [Troubleshooting](https://hexo.io/docs/troubleshooting.html).
- [X] Yes, I was on [NexT Documentation Site](http://theme-next.org/docs/).
- [X] And yes, I already searched for current [issues](https://github.com/theme-next/hexo-theme-next/issues?utf8=%E2%9C%93&q=is%3Aissue) and this is not help to me.

***

### Expected behavior 
My site is running in a subdirectory, for an instance: xxx.github.io/en
Every link of post title should be something like xxx.github.io/en/posts/{Post-Title}, and every read more button should have the same link as the post title.

### Actual behavior 
After I update the latest release of the theme, every link of post title becomes xxx.github.io/en/en/posts/{Post-Title}, so that's a duplicate of my subdirectory name and caused 404. However, the read more button still has the right link to a post like xxx.github.io/en/posts/{Post-Title}

This is a part of my hexo configuration:
``` YAML
# URL
## If your site is put in a subdirectory, set url as 'http://yoursite.com/child' and root as '/child/'
url: https://bitrhythm.github.io/en
root: /en/
permalink: posts/:title/
permalink_defaults:
```

This is the screenshot:
![c](https://user-images.githubusercontent.com/23711687/48303184-18b64200-e542-11e8-8987-1ddc2317fd34.PNG)

This is the html source:
``` HTML
Post title link:
<a href="/en/en/posts/Hello-World/" class="post-title-link" itemprop="https://bitrhythm.github.io/en/index.html">Hello World</a>

Read more link:
<a class="btn" href="/en/posts/Hello-World/#more" rel="contents"> Read more » </a>
```

### Steps to reproduce the behavior 
1. N/A
2. N/A
3. N/A

* Link to demo site with this issue: N/A
* Link(s) to source code or any usefull link(s): N/A

### Node.js and NPM Information

```

```

### Package dependencies Information

```

```

#### Hexo version

```

```

#### Hexo Configuration

```yml

```

### NexT Information

**NexT Version:**

- [ ] Latest Master branch
- [ ] Latest Release version
- [ ] Old version

**NexT Scheme:**

- [ ] All schemes
- [ ] Muse
- [ ] Mist
- [ ] Pisces
- [ ] Gemini

#### NexT Configuration:

```yml

```

**Comment by ivan-nginx** [comment]
`root: /en/` – not need.

**Comment by Bitrhythm** [comment]
> `root: /en/` – not need.

```
Website in subdirectory
If your website is in a subdirectory (such as http://example.org/blog) set url to http://example.org/blog and set root to /blog/.
```

But this is what the hexo docs says. And the read more button actually has the right link, so I'd rather believe this is not the real cause of the problem.

**Comment by ChaosRinkHan** [comment]
I've also got this problem in updating to NexT 6 from NexT 5.1.3. 
In NexT 5 the post links are correctly generated, but with the same configuration in NexT 6 links are generated wrongly with subdirectory repeating twice as described above.

**Comment by JiangTJ** [comment]
@sli1989 
This may be an issue
![image](https://user-images.githubusercontent.com/15902347/48399116-b2a60680-e75d-11e8-95ab-209ab7a2e497.png)
![image](https://user-images.githubusercontent.com/15902347/48399210-031d6400-e75e-11e8-973a-be4408dc09ea.png)
`url_for()` is used twice.

**Comment by ivan-nginx** [comment]
Yes, please, check it. May be bug.

**Comment by Bitrhythm** [comment]
I changed ```var attrs = { href: this.url_for(path) };``` to ```var attrs = { href: path };``` in **scripts/helpers/next-url.js**, and then the problem solved...thanks to @JiangTJ
But I'm not sure this is the right solution, so @ivan-nginx please check it.

**Comment by ivan-nginx** [comment]
Ok, i'll check it later. Please, wait. And thank's for report.

**Comment by stevenjoezhang** [comment]
It seems the same problem with #231, caused by `url_for()`.

**Comment by Bitrhythm** [comment]
Update: I tried changing ```relative_link: false -> true``` in hexo's _config.yml file, this DID solve the problem of the post link but caused a new problem that ALL NAVIGATION LINK becomes unclickable, in other words, all navigation link tag lost its value of href attribute.
Hope this gave you guys helpful information.

**Comment by bryfroze** [comment]
> I changed `var attrs = { href: this.url_for(path) };` to `var attrs = { href: path };` in **scripts/helpers/next-url.js**, and then the problem solved...thanks to @JiangTJ
> But I'm not sure this is the right solution, so @ivan-nginx please check it.
> 
> Update: this solution will cause all navigations links to lose their subdirectory name, /subdirectory/categories become /categories

Same issue, change `attrs` can not solve problem

**Comment by bryfroze** [comment]
> I changed `var attrs = { href: this.url_for(path) };` to `var attrs = { href: path };` in **scripts/helpers/next-url.js**, and then the problem solved...thanks to @JiangTJ
> But I'm not sure this is the right solution, so @ivan-nginx please check it.
> 
> Update: this solution will cause all navigations links to lose their subdirectory name, /subdirectory/categories become /categories

I fix that, seems alright.

**Comment by ivan-nginx** [comment]
So, this bug only for subdirectory? If will not subdirectory, fix will work fine?

**Comment by Bitrhythm** [comment]
Yes, this bug shouldn't appear if site were not in subdirectory.

**Comment by JiangTJ** [comment]
@ivan-nginx 
![image](https://user-images.githubusercontent.com/15902347/49557927-3c538900-f944-11e8-81f7-0f0d014d678f.png)
![image](https://user-images.githubusercontent.com/15902347/49557959-568d6700-f944-11e8-954e-5f7f33ec0571.png)
Some use `next_url(url_for())`, but some `next_url()`, it maybe reason.
commint sha:819e1a5d38fa1a1e9a4f5a5ac10e57530d339b36

**Comment by ivan-nginx** [comment]
Maybe need to use just `next_url()`?

**Comment by stevenjoezhang** [comment]
Should be fixed in #496
